### PR TITLE
Extend types

### PR DESCRIPTION
**Pull request by xlarsx**
- Extend the generic usage
- Added missing return types
- `Delete` behavior consistent with the others
- Explicit `BaseDAO` defined as `Abstract`

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -6,53 +6,37 @@ export = BaseDAO;
...
-  formatData(rawData: object): Promise<object>;
-  getFiledName(key: string, addTable: string): string;
-  getErrorHandler(actionName: string): Function;
-  formatFields(customFields?: string[]): string[];
 }
 
 declare module BaseDAO {
-  export interface IBaseDAO {
+  export interface IBaseDAO<T extends object> {
     dbTable: string;
```
Comment (itereshchenkov) [inline index.d.ts]: can you please also add `fields` property?

Comment (xlarsx) [reply index.d.ts]: ready, thanks for the feedback

File: index.d.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ type ErrorHandler = (err: Error, actionName: string) => void;
 
 declare class BaseDAO<T extends object> implements BaseDAO.IBaseDAO<T> {
   public dbTable: string;
+  private fields: object;
```
Comment (itereshchenkov) [inline index.d.ts]: it's better to set is as `protected`, as we can override it in a child class, e.g. to override caster
https://github.com/collectai/landing-page/blob/master/src/db/dao/PageData.js#L14

Comment (xlarsx) [reply index.d.ts]: agree, fixed

**Pull request merged**